socket.c

```diff
@@ -1909,10 +1989,7 @@ int camblet_connect(struct sock *sk, struct sockaddr *uaddr, int addr_len)
 
 error:
 	camblet_socket_free(sc);
-
-	lock_sock(sk);
-	sk->sk_prot->close(sk, 0);
-	release_sock(sk);
+	socket_reset(sk);
```

Comment:
I think we should lock the socket during the reset phase.

Reply:
the lock/release were intentionally removed, my current understanding is that the socket is already locked in the function that called `camblet_connect`